pkg/storage/queue.go

```diff
@@ -521,17 +521,28 @@ func (bq *baseQueue) processReplica(
 	defer cancel()
 	log.Eventf(ctx, "processing replica")
 
+	if destroyed := func() bool {
+		repl.mu.Lock()
+		defer repl.mu.Unlock()
+		return repl.mu.destroyed != nil
```

Comment:
This kind of logic is best implemented within replica.go . I don't have a good suggestion but can it be implemented within redirectOnOrAcquireLease() ?